extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisRecordProcessor.scala

```diff
@@ -135,18 +105,17 @@ private[kinesis] class KinesisRecordProcessor[T](
    */
   override def shutdown(checkpointer: IRecordProcessorCheckpointer, reason: ShutdownReason) {
     logInfo(s"Shutdown:  Shutting down workerId $workerId with reason $reason")
-    reason match {
+    // We want to return a checkpointer based on the shutdown cause. If we want to terminate,
+    // then we want to use the given checkpointer to checkpoint one last time. In other cases,
+    // we return a `null` so that we don't checkpoint one last time.
+    val cp: IRecordProcessorCheckpointer = reason match {
```

Comment:
Can this code be just simplified to 
```
reason match {
  case ShutdownReason.TERMINATE => receiver.removeCheckpointer(shardId, cp)
  case _ =>  receiver.removeCheckpointer(shardId, null)   // return null so that we don't checkpoint 
}